docs/okeps/okep-5085-localnet-api.md

```diff
@@ -0,0 +1,596 @@
+    // Topology describes network configuration.
+    //
+    // Allowed values are "Layer3", "Layer2" and "Localnet".
+    // Layer3 topology creates a layer 2 segment per node, each with a different subnet. Layer 3 routing is used to interconnect node subnets.
+    // Layer2 topology creates one logical switch shared by all nodes.
+    // Localnet topology attach to the nodes physical network. Enables egress to the provider's physical network. 
+    //
+    // +kubebuilder:validation:Required
+    // +required
+    // +unionDiscriminator
+    Topology NetworkTopology `json:"topology"`
+    ...
+    // Localnet is the Localnet topology configuration.
+    // +optional
+    Localnet *LocalnetConfig `json:"localnet,omitempty"`
+}
+```
+
+The CUDN spec should have additional validation rule for `spec.network.topology` field:
+```go
+// ClusterUserDefinedNetworkSpec defines the desired state of ClusterUserDefinedNetwork.
+type ClusterUserDefinedNetworkSpec struct {
+    ...
+    // +required
+    Network NetworkSpec `json:"network"`
+}
+```
+
+#### Localnet topology configuration type
+Introduce new topology configuration type for localnet - `LocalnetConfig`.
+The CUDN CRD `spec.network` should feature proposed localnet topology configuration.
+
+The Layer2 and Layer3 configuration types (`Layer2Config` & `Layer3Config`) `role` field is defined with `NetworkRole` type.
+The `NetworkRole` type has the following enum validation, allowing `Secondary` and `Primary` values:
+```
+// +kubebuilder:validation:Enum=Primary;Secondary
+```
+The proposed localnet config type (`LocalnetConfig`) `role` field would have to accept `Secondary` value only.
+In order to avoid misleading CRD scheme, the enum validation should be moved closer to each `NetworkRole` usage:
+1. Remove the enum validation exist on `NetworkRole` definition
+2. At the `Layer2Config` definition, add enum validation to `role` field allowing: `Primary` or `Secondary`.
+3. At the `Layer3Config` definition, add enum validation to `role` field allowing: `Primary` or `Secondary`.
+4. At the proposed `LocalnetConfig` definition, add enum validation to `role` field allowing `Secondary` only.
+
+#### VLAN configuration type
+As of today OVN-Kubernetes CNI enable setting the access VLAN of a localnet topology the network (using NADs).
+
+There are two options to expose the VLAN attribute:
+1. Single integer filed, living at the same level as other fields, for example:
+   ```
+    localnet:
+       mtu: 1500
+       vlan: 100
+   ```
+2. Discriminated union for accommodating VLAN related configurations, for example: 
+   ```
+    localnet:
+       mtu: 1500
+       vlan: 
+         mode: Access
+         access:
+           id: 100 
+   ```
+
+Although option (1) being pretty straight forward, in case additional VLAN configurations should be supported requiring 
+exposing additional fields, it will enforce having VLAN related filed right next to non VLAN related ones, making the API look occurred.
```

Comment:
```suggestion
the exposure of additional fields, it will enforce having VLAN-related fields right next to non-VLAN-related ones, making the API look awkward.
```